server/src/controllers/sync.controller.ts

```diff
@@ -24,4 +37,37 @@ export class SyncController {
   getDeltaSync(@Auth() auth: AuthDto, @Body() dto: AssetDeltaSyncDto): Promise<AssetDeltaSyncResponseDto> {
     return this.service.getDeltaSync(auth, dto);
   }
+
+  @Post('stream')
+  @Header('Content-Type', 'application/jsonlines+json')
+  @HttpCode(HttpStatus.OK)
+  @Authenticated()
+  async getSyncStream(@Auth() auth: AuthDto, @Res() res: Response, @Body() dto: SyncStreamDto) {
+    try {
+      await this.service.stream(auth, res, dto);
+    } catch (error: Error | any) {
+      res.setHeader('Content-Type', 'application/json');
+      this.errorService.handleError(res, error);
+    }
```

Comment:
It would be nice if that handling was in the service but I'm assuming errors can be thrown in too many places to be feasible?

Reply:
I don't rent http specific stuff in the service. This isn't "handling" the error as much as it is converting it to an http response which doesn't belong in the service.